observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,194 @@
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.UnmodifiableListIterator;
+import io.grpc.MethodDescriptor;
+import io.grpc.observability.ObservabilityConfig;
+import io.grpc.observability.ObservabilityConfig.LogFilter;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ConfigFilterHelper {
+  private static final Logger logger = Logger.getLogger(ConfigFilterHelper.class.getName());
+
+  private static final MethodFilterParams NO_FILTER_PARAMS
+      = new MethodFilterParams(false, 0, 0);
+
+  private final ObservabilityConfig config;
+  boolean methodOrServiceFilterPresent;
+  boolean globalLog;
+  MethodFilterParams globalParams;
+  Map<String, MethodFilterParams> perServiceFilters;
+  Map<String, MethodFilterParams> perMethodFilters;
+  Set<EventType> logEventTypeSet;
+
+  @VisibleForTesting
+  ConfigFilterHelper(ObservabilityConfig config) {
+    this.config = config;
+  }
+
+  /**
+   * Creates and returns helper instance for log filtering.
+   *
+   * @param config processed ObservabilityConfig object
+   * @return helper instance for filtering
+   */
+  public static ConfigFilterHelper getInstance(ObservabilityConfig config) {
+    ConfigFilterHelper filterHelper = new ConfigFilterHelper(config);
+    filterHelper.setEmptyFilter();
+    if (config.isEnableCloudLogging()) {
+      filterHelper.setMethodOrServiceFilterMaps();
+      filterHelper.setEventFilterSet();
+    }
+    return filterHelper;
+  }
+
+  @VisibleForTesting
+  void setEmptyFilter() {
+    this.globalLog = false;
+    this.globalParams = NO_FILTER_PARAMS;
+    this.methodOrServiceFilterPresent = false;
+    this.perServiceFilters = new HashMap<>();
+    this.perMethodFilters = new HashMap<>();
+    this.logEventTypeSet = ImmutableSet.of();
+    return;
+  }
+
+  @VisibleForTesting
+  void setMethodOrServiceFilterMaps() {
+    ImmutableList<LogFilter> logFilters = config.getLogFilters();
+    if (logFilters == null) {
+      return;
+    }
+
+    boolean globalLog = false;
+    MethodFilterParams globalParams = null;
+    Map<String, MethodFilterParams> perServiceFilters = new HashMap<>();
+    Map<String, MethodFilterParams> perMethodFilters = new HashMap<>();
+
+    UnmodifiableListIterator<LogFilter> logFilterIterator = logFilters.listIterator();
+    while (logFilterIterator.hasNext()) {
+      // '*' for global, 'service/*' for service glob, or 'service/method' for fully qualified
+      LogFilter currentFilter = logFilterIterator.next();
+      String methodOrServicePattern = currentFilter.pattern;
+      int currentHeaderBytes
+          = currentFilter.headerBytes != null ? currentFilter.headerBytes : 0;
+      int currentMessageBytes
+          = currentFilter.messageBytes != null ? currentFilter.messageBytes : 0;
+      if (methodOrServicePattern.equals("*")) {
```

Comment:
This can also occur more than once i.e. if `globalLog` is already set. So the 2 things that are different (see below) are you don't generate the warning saying "duplicate entry" and the later one overrides the previous one.

Reply:
Added duplicate entry check to`globalLog` as well, so that parameters do not get overrided if same pattern is found later.